tests/filters/heuristic_filter_test.py

```diff
@@ -90,10 +91,15 @@ def test_success(self, secret, line):
 
             # fail although the word david ends in id
             ('RANDOM_STRING', 'postgres://david:RANDOM_STRING'),
+
+            # fail since this is an aws access key id, a real secret
+            ('AKIA4NACSIJMDDNSEDTE', 'aws_access_key_id=AKIA4NACSIJMDDNSEDTE'),
         ],
     )
-    def test_failure(self, secret, line):
-        assert not filters.heuristic.is_likely_id_string(secret, line)
+    def test_failure(self, secret, line, plugin=None):
+        if secret.startswith('AKIA'):
+            plugin = AWSKeyDetector()
```

Comment:
I'm not the biggest fan of the custom code here. Can we possibly add this to the parameterized inputs?